components/Chat.tsx

```diff
@@ -63,34 +71,33 @@ export default function Chat() {
         throw new Error(`HTTP error! status: ${res.status}`);
       }
 
-      const { id } = await res.json()
+      const { id } = await res.json();
 
       const interval = window.setInterval(async () => {
         const pollRes = await fetch(`/api/chat/poll?job_id=${id}`, {
-          method: 'GET',
-        })
+          method: "GET",
+        });
         if (!res.ok) {
           setLoading(false);
-          console.error("Stopped polling")
-          window.clearInterval(interval)
-          return
+          console.error("Stopped polling");
+          window.clearInterval(interval);
+          return;
         }
 
-        const { data, completed } = await pollRes.json()
+        const { data, completed } = await pollRes.json();
         if (completed) {
           addMessage({
             role: "system",
             content: data,
-          })
+          });
           setLoading(false);
           // clear interval and return
-          window.clearInterval(interval)
-          return
+          window.clearInterval(interval);
+          return;
         }
-      }, 1000)
-
+      }, 1000);
     } catch (error) {
-      console.error('Error:', error);
+      console.error("Error:", error);
     }
```

Comment:
In the event of a polling error can we show a toast or an alert and reset the UI state? given the state of our code if we do deploy it is probably likely users are going to see errors, we should try to warn them something has happened (not just in the console)

Idea: have an alert context in a provider from the root component that passes down a `QueueAlert` function which adds alerts to the queue. Then any component which receives an error can queue up alerts and the root component will render them as toasts. Maybe 3-5 seconds before selecting the next queued alert, or only show the next queued alert on closing the current one

Reply:
Thanks, I added a toast when an error occurs using Material UI toolpad's notification provider. It queues notifications and shows a badge with the number of queued notifications
<img width="695" alt="Screenshot 2024-11-30 at 6 38 05 PM" src="https://github.com/user-attachments/assets/603b71a6-a945-49ba-b2ed-eb0a6b9c33a6">